diff --git a/src/Plugin/search_api/processor/BoostMoreRecent.php b/src/Plugin/search_api/processor/BoostMoreRecent.php
index 236f8f0..6ab12d0 100644
--- a/src/Plugin/search_api/processor/BoostMoreRecent.php
+++ b/src/Plugin/search_api/processor/BoostMoreRecent.php
@@ -115,7 +115,7 @@ class BoostMoreRecent extends ProcessorPluginBase implements PluginFormInterfac
    */
   public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
     $values = $form_state->getValues();
-    foreach ($values['boosts'] as $field_id => $boost) {
+    foreach (($values['boosts'] ?? []) as $field_id => $boost) {
       if (Utility::formatBoostFactor(0) === $boost['boost']) {
         unset($values['boosts'][$field_id]);
       }
